code/edge/src/main/java/com/adobe/marketing/mobile/EdgeHitProcessor.java

```diff
@@ -276,6 +197,159 @@ public void onComplete() {
 		}
 	}
 
+	/**
+	 * Process and send an ExperienceEvent network request.
+	 *
+	 * @param entityId the {@link DataEntity} unique identifier
+	 * @param entity the {@link EdgeDataEntity} which encapsulates the request data
+	 * @param request a {@link RequestBuilder} instance
+	 * @return true if the request processing is complete for this hit or false if processing is
+	 * not complete and this hit must be retired.
+	 */
+	private boolean processExperienceEventHit(
+		@NonNull final String entityId,
+		@NonNull final EdgeDataEntity entity,
+		@NonNull final RequestBuilder request
+	) {
+		if (stateCallback != null) {
+			// Add Implementation Details to request (global) level
+			request.addXdmPayload(stateCallback.getImplementationDetails());
+		}
+
+		final List<Event> listOfEvents = new ArrayList<>();
+		listOfEvents.add(entity.getEvent());
+		final JSONObject requestPayload = request.getPayloadWithExperienceEvents(listOfEvents);
+
+		if (requestPayload == null) {
+			Log.warning(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Failed to build the request payload, dropping current event (%s).",
+				entity.getEvent().getUniqueIdentifier()
+			);
+
+			return true; // Request complete, don't retry hit
+		}
+
+		Map<String, Object> edgeConfig = entity.getConfiguration();
+
```

Comment:
Should we add method to entity to get edgeConfigId directly? 

entity.getConfiguration.getEdgeConfigId()